Docu update (#562)
smart-data-lake/smart-data-lake: docs/reference/transformations.md

---

:::warning
This page is under review 
:::

## Spark Transformations
To implement custom transformation logic, specify **transformers** attribute of an Action. 

> It allows you to chain several transformations in a linear process, where output SubFeeds from one transformation are use as input for the next.  

Reviewer: ```suggestion
It allows you to chain several transformations in a linear process, where the output SubFeeds of one transformation are used as input for the next.
```

Depending on your Action type the transformations have different format (described later). The two types are:

* **1-to-1** transformations (\*DfTransformer): One input DataFrame is transformed into one output DataFrame. This is the case for *CopyAction*, *DeduplicateAction* and *HistorizeAction*.
* **many-to-many** transformations (\*DfsTransformer): Many input DataFrames can be transformed into many output DataFrames. This is the case for *CustomDataFrameAction*.

The configuration allows you to use [*predefined standard transformations*](#predefined-transformations) or to define [*custom transformation*](#custom-transformations) in various languages.

:::warning Deprecation Warning
 there has been a refactoring of transformations in Version 2.0.5. The attribute **transformer** is deprecated and will be removed in future versions. Use **transformers** instead.
:::

### Predefined Transformations
Predefined transformations implement generic logic to be reused in different actions. 

> Depending on the transformer there are a couple of option to specify properties, see [Configuration Schema Viewer](../../JsonSchemaViewer).

Reviewer: ```suggestion
Depending on the transformer there are a couple of properties you can specify, see [Configuration Schema Viewer](../../JsonSchemaViewer).
```
